scripts/filter_expected_warnings.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+"""Script to filter out old warnings from doxygen."""
+
+import sys
+import re
+
+
+class DoxygenWarning(object):
+    """Doxygen warning class."""
+
+    def __init__(self, firstline, filename, warning):
+        self.firstline = firstline
+        self.filename = filename
+        self.warning = warning
+        self.otherlines = []
+
+    def approx_equals(self, other):
+        """Return true if warnings have same filename and warning message."""
+        if self.filename != other.filename:
+            return False
+        if self.warning != other.warning:
+            return False
+        return self.otherlines == other.otherlines
+
+    def is_in_list(self, warnings_list):
+        """Check if this (or a similar warning) is in the list."""
+        for other in warnings_list:
+            if self.approx_equals(other):
+                return True
+        return False
+
+    def print_with_prefix(self, prefix):
+        print(prefix + self.firstline)
+        for line in self.otherlines:
+            print(prefix + '  ' + line)
+
+
+def build_warnings_list(all_lines):
+    """Create a list of the warnings in this file."""
+    warnings_list = []
+    current = None
+
+    warning_start_expr = re.compile(r'[^:]+/([^:/]+):\d+: (warning.*$)')
+
+    for line in all_lines:
+        if line.isspace():
+            continue
+
+        # Allow comments in the list of expected warnings.
+        if line.startswith('#'):
+            continue
+
+        matched = warning_start_expr.match(line)
+        if matched:
+            filename = matched.group(1)
+            warning = matched.group(2)
+            current = DoxygenWarning(line.strip(), filename, warning)
+            warnings_list.append(current)
+        elif line.startswith('  '):
+            current.otherlines.append(line.strip())
+        else:
+            # Assuming all warnings are of the form [path:line: warning:...]
+            # (and the warnings about too many nodes have been filtered out).
+            print('Error filtering warnings: Unexpected input format.')
+            print('  Input:' + line)
+
+    return warnings_list
+
+
+def ignore_too_many_nodes(all_lines):
+    """Filter out lines about graphs with too many nodes."""
+    too_many_nodes_expr = re.compile(
+        r'warning: Include(d by)? graph for .* not generated, too many nodes. '
+        + r'Consider increasing DOT_GRAPH_MAX_NODES.')
+    return [x for x in all_lines if not too_many_nodes_expr.match(x)]
+
+
+def list1_is_in_list2(list1, list2, prefix):
```

Comment:
```suggestion
def subset(superset, subset, prefix):
```

And then when you call it

```
subset(superset=some_list, subset=some_other_list)
```